komodo/build.py

```diff
@@ -245,16 +252,8 @@ def make(
     pip="pip",
     fakeroot=".",
 ):
-    packages_needed = flatten(dfs(pkg, ver, pkgs, repo) for pkg, ver in pkgs.items())
-
-    seen_package = set()
-    pkgorder = []
-    for package_needed in packages_needed:
-        if package_needed in seen_package:
-            continue
-        seen_package.add(package_needed)
-        pkgorder.append(package_needed)
-
+    pkgorder = full_dfs(pkgs, repo)
+    assert len(set(pkgorder)) == len(pkgorder) == len(pkgs)
```

Comment:
```
assert len(set(pkgorder)) == len(pkgs)
```
Will this not suffice? The set of pkgorder will be unique, and is bounded by len(pkgs) regardless?